subprojects/workers/src/main/java/org/gradle/workers/internal/AbstractWorker.java

```diff
@@ -51,11 +51,20 @@ public DefaultWorkResult call(BuildOperationContext context) {
             public BuildOperationDescriptor.Builder description() {
                 return BuildOperationDescriptor.displayName(spec.getDisplayName())
                     .parent(parentBuildOperation)
-                    .details(new Details(spec.getImplementationClass().getName(), spec.getDisplayName()));
+                    .details(new Details(getImplementationClassName(spec), spec.getDisplayName()));
             }
         });
     }
 
+    private static String getImplementationClassName(ActionExecutionSpec spec) {
+        if (spec.getImplementationClass() == AdapterWorkerExecution.class) {
```

Comment:
did you intend to check for identity here?

Reply:
As opposed to what?  `equals()`?  The implementation for `Class.equals()` is:
```
public boolean equals(Object obj) {
        return (this == obj);
}
```

Or am I misunderstanding the question?